packages/api/src/controllers/stream.ts

```diff
@@ -1677,7 +1710,10 @@ app.patch(
     const stream = await db.stream.get(id);
 
     const exists = stream && !stream.deleted;
-    const hasAccess = stream?.userId === req.user.id || req.isUIAdmin;
+    const hasAccess =
+      stream?.userId === req.user.id ||
+      (stream?.projectId ?? "" !== req.project?.id ?? "") ||
+      req.isUIAdmin;
```

Comment:
You're mixing up `!==` and `===` checks here. I gues sit should be `===` and `&&`. (my helper function suggestion might help with these)